### ISSUE TITLE
Make Plonk's `verify` in-circuit

### ISSUE THREAD

**Issue by vlasin** [zkFold/zkfold-base#144]

**Comment by TurtlePU** [comment]
So I guess this can be done since @echatav's work on #396 is done?

**Comment by vlasin** [comment]
Yes, although it is not a priority now. We should first be able to compile BLS scalar multiplication before attempting this.
